sound/hybrid_font.h

```diff
@@ -207,17 +218,35 @@ class HybridFont : public SaberBase {
               swing_player_ = PlayPolyphonic(&SFX_swing);
             }
             swinging_ = true;
+          } else {
+            if (angle_ > font_config.ProffieOSSpinDegrees) {
+              if (SFX_spin) {
+                swing_player_ = PlayPolyphonic(&SFX_spin);
+              } else if (SFX_slsh) {
+                swing_player_ = PlayPolyphonic(&SFX_slsh);
+              } else if (SFX_swng) {
+                swing_player_ = PlayPolyphonic(&SFX_swng);
+              } else {
+                swing_player_ = PlayPolyphonic(&SFX_swing);
+              }
+              angle_ -= font_config.ProffieOSSpinDegrees;
+            }
           }
         }
       } else if (!swinging_ && swing_speed > swingThreshold) {
         PlayMonophonic(&SFX_swing, &SFX_hum);
         swinging_ = true;
+        if (angle_ > 360 && swinging_) {
+          PlayMonophonic(&SFX_spin, &SFX_hum);
+          angle_ -= font_config.ProffieOSSpinDegrees;
+        }
       }
       float swing_strength = std::min<float>(1.0, swing_speed / swingThreshold);
       SetSwingVolume(swing_strength, 1.0);
     } else if (swing_speed <= font_config.ProffieOSSwingLowerThreshold) {
       swinging_ = false;
       swing_player_.Free();
+      angle_ = 0;
```

Comment:
By doing this, any drift we have should go away when we're not swinging.

Reply:
agreed, tested fine.